ocaml/xapi/xapi_xenops.ml

```diff
@@ -667,10 +667,16 @@ module MD = struct
                   |> List.assoc Xapi_globs.vgpu_type_id in
     let uuid = vgpu.Db_actions.vGPU_uuid in
     let extra_args = vgpu.Db_actions.vGPU_extra_args in
+    let config_file = match Xapi_vgpu_type.Vendor_nvidia.is_host_driver_support_multi_vgpu () with
```

Comment:
We can avoid the call to this function here (where it is called very often), and move it into `create_compat_config_file` (which is called just once, when xapi starts). I.e. only add the `config_file` key to the `internal_config` map if needed.

Then over here you just do something like
```
let config_file =
  if List.mem_assoc Xapi_globs.nvidia_compat_config_file_key internal_config then
    Some (List.assoc Xapi_globs.nvidia_compat_config_file_key internal_config)
  else
    None
in
```

Reply:
oh, I see.

Reply:
Oh, there is now a `List.assoc_opt` function in the standard lib. Then it becomes just:
```
let config_file = List.assoc_opt Xapi_globs.nvidia_compat_config_file_key internal_config in
```